internal/pkg/cli/deploy/workload.go

```diff
@@ -230,8 +232,12 @@ func newWorkloadDeployer(in *WorkloadDeployerInput) (*workloadDeployer, error) {
 	}
 
 	repoName := fmt.Sprintf("%s/%s", in.App.Name, in.Name)
-	imageBuilderPusher := repository.NewWithURI(
+	repository := repository.NewWithURI(
 		ecr.New(defaultSessEnvRegion), repoName, resources.RepositoryURLs[in.Name])
+	uri, loginOut, err := repository.Login()
+	if err != nil {
+		return nil, fmt.Errorf("login to docker %s: %w", loginOut, err)
```

Comment:
could we move the `Login()` call to `uploadContainerImages()`? then we don't have to save the docker login output on `workloadDeployer`!

Reply:
So if we move `Login() to the `uploadContainerImages()` function, then we perform every time `Login` even when we upload custom resources, EnvFiles` Is this fine?

Reply:
i think `uploadContainerImages()` only gets called once? sequentially to when we upload env files and custom resources? where are you seeing that it'll get called multiple times?

Reply:
I mean if there are no container images to upload but we have env files to upload then also we perform `Login()`.

Reply:
So I have the below code

 ```
func (d *workloadDeployer) uploadContainerImages(out *UploadArtifactsOutput) error {
	uri, dockerLoginOutput, err := d.repository.Login()
	if err != nil {
		return fmt.Errorf("login to docker: %w. `docker login` output: %s", err, dockerLoginOutput)
	}
	// If it is built from local Dockerfile, build and push to the ECR repo.
	buildArgsPerContainer, err := buildArgsPerContainer(d.name, d.workspacePath, uri, d.image, d.mft)
```

So I am performing Login just before the `buildArgsPerContainer()` to pass `URI ` as an argument to set `buildArgs`. As a result, the `Login` method is executed even when there are no container images to upload.

Reply:
ooooh i see what you're saying. i actually think we want the new behavior of only logging in if we're going to upload containers. The `Login()` function logs in the local `docker` engine in to ECR, which we only need if we're pushing container images there!

Reply:
Done! moved the `Login` to `uploadContainerImages`